### PR TITLE
Allow saving composition notes

### LINKED ISSUE #32: Support Markdown formatting in the 'notes' text field
There are a few parts to supporting Markdown in the 'notes' field:

- When we render the notes on a static page where the user can't edit them, we should convert the Markdown to HTML so that links, boldface, and other formatting appears.
- In the composition edit form, it would be cool to offer a toolbar so the user doesn't have to manually type Markdown if they don't want to. This isn't critical, but maybe there's a simple drop-in solution?
    new_issue_ _cheshire137_overwatch-team-comps
- We should strip certain kinds of formatting from the 'notes' field, either on render or before we save the value in the database. `style` attributes, `img` tags, `embed` tags, etc. we don't want to render.

### PR DESCRIPTION
**Pull request by cheshire137**
This fixes #32:

- You can now save notes about your team comp.
- You can paste in links to Gfycat, Twitch, YouTube, and Streamable videos and they will be automatically embedded.
- The notes field grows as you add new lines.
- You can use basic Markdown to add custom links, headers, boldface/italic/underline, etc. formatting to your notes.

**The editor:**

team_composition_form_-_overwatch_team_comps

----

**The result:**

clearly_the_best_-_hanamura_-_overwatch_team_comps

/cc @RobThePM

### CODE REVIEW

File: app/assets/javascripts/components/composition-notes.jsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    this.setState({ notes: event.target.value })
+  }
+
+  saveNotes(event) {
+    if (this.state.isRequestOut) {
+      return
+    }
+    if (event) {
+      event.target.blur()
+    }
```
Comment (rewinfrey) [inline app/assets/javascripts/components/composition-notes.jsx]: @cheshire137 I'm not understanding why this `blur` is required and I'm curious :) Mind explaining this to me?

What I understand now is the only time we have an event passed to `this.saveNotes` is when the `save-notes-button` is clicked. The target of the event is the button, and so I'm not sure what `blur` adds in that case?

Comment (cheshire137) [inline app/assets/javascripts/components/composition-notes.jsx]: The blur is just to remove the border usually added to a clicked/focused button. So it goes from something like this:

allow_saving_composition_notes_by_cheshire137_ _pull_request__136_ _cheshire137_overwatch-team-comps

To this:

allow_saving_composition_notes_by_cheshire137_ _pull_request__136_ _cheshire137_overwatch-team-comps

So it's appearance only, doing the blur.

Comment (rewinfrey) [inline app/assets/javascripts/components/composition-notes.jsx]: @cheshire137 ah, thanks! I forgot buttons get a border for blur. 👍

**Comment by RobThePM** [comment]
Thanks Chesh! And Rick! Sorry for not responding to this earlier.

**Pull request merged**